block/s3/inventory.go

```diff
@@ -125,5 +127,43 @@ func sortManifest(m *Manifest, logger logging.Logger, reader s3inventory.IReader
 			return ErrInventoryFilesRangesOverlap
 		}
 	}
+	filteredFiles := make([]inventoryFile, 0, len(m.Files))
+	if len(prefixes) == 0 {
+		return nil
+	}
+	// filter manifest files according to prefixes:
+	sort.Strings(prefixes)
+	currentPrefixIdx := 0
+	defer func() {
+		m.Files = filteredFiles
```

Comment:
Mutating state in a `defer`ed function makes it harder to follow and debug IMO. it's basically a GOTO where the logical flow of the program jumps to a different location...